internal/manifest/oci_index.go

```diff
@@ -102,13 +115,26 @@ func (index *OCI1IndexPublic) editInstances(editInstances []ListEdit) error {
 				return fmt.Errorf("update %d of %d passed to OCI1Index.UpdateInstances had no media type (was %q)", i+1, len(editInstances), index.Manifests[i].MediaType)
 			}
 			index.Manifests[targetIndex].MediaType = editInstance.UpdateMediaType
+			for k, v := range editInstance.UpdateAnnotations {
+				index.Manifests[targetIndex].Annotations[k] = v
+			}
+			annotations := make(map[string]string)
+			addCompressionAnnotations(editInstance.UpdateCompressionAlgorithms, annotations)
+			for k, v := range annotations {
+				index.Manifests[targetIndex].Annotations[k] = v
+			}
```

Comment:
`addCompressionAnnotations(…, index.Manifests[…].Annotations)` would AFAICS do the same thing in a simpler way; no need for an extra zero-or-one-element map.

If you think the effect is not clear, `addCompressionAnnotations(*Descriptor, []Alghoritm)` would make it very clear that it is editing the descriptor.